File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.java
// Last updated: 8/1/2025, 6:26:36 PM
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int count = 0; //intitialize count
        int result = 0; //initialize max
      
        for (int i=0 ; i < nums.length ; i++)
        {
              
            // Reset count when 0 is found
            if (nums[i] == 0)
                count = 0;
      
            // If 1 is found, increment count
            // and update result if count becomes
            // more.
            else
            {
                count++;//increase count
                result = Math.max(result, count);
            }
        }
      
        return result;
        
    }
}

File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.py
# Last updated: 8/1/2025, 6:26:33 PM
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        res = 0
        count = 0
        for n in nums:
            if n == 1:
                count += 1
            else:
                count = 0
        
            res = max(count, res)
        return res
